anchore_engine/services/policy_engine/engine/vulns/cpe_matchers.py

```diff
@@ -0,0 +1,330 @@
+from typing import List, Tuple
+
+from sqlalchemy.orm import Session
+from collections import namedtuple
+
+from anchore_engine.db import (
+    NvdV2Metadata,
+    CpeV2Vulnerability,
+    Image,
+    get_thread_scoped_session,
+    ImageCpe,
+    VulnDBCpe,
+    Vulnerability,
+    DistroNamespace,
+    ImagePackage,
+)
+from anchore_engine.services.policy_engine.engine.vulns.cpes import (
+    FuzzyCandidateCpeGenerator,
+    dedup_cpe_vulnerabilities,
+)
+from anchore_engine.services.policy_engine.engine.vulns.db import (
+    CpeDBQueryManager,
+    db_result_tuples_to_list,
+)
+from anchore_engine.subsys import logger
+from anchore_engine.utils import timer
+
+# Match class to bind the image and vuln sides and ease comparisons and hashing
+CpeMatch = namedtuple("CpeMatch", ["image_cpe", "vuln_cpe"])
+
+
+def cve_ids_for_vuln_record(vuln: Vulnerability) -> List[str]:
```

Comment:
is this type hint for the method return correct?

Reply:
seems like it should be List[set(str)]

Reply:
that is a constructor call to create a list from a set from the list (basically does a dedup pass), but the result is a list of strings.

Reply:
```
l1 = [1,2,3,3,4]
list(set(l1))
[1, 2, 3, 4]
```